app/src/main/java/io/gnosis/safe/ui/assets/coins/CoinsViewModel.kt

```diff
@@ -45,17 +47,21 @@ class CoinsViewModel
                     )
                 }
                 val balanceInfo = tokenRepository.loadBalanceOf(safe.address, userDefaultFiat)
-                val balances = getBalanceViewData(balanceInfo)
-
-                updateState { CoinsState(loading = false, refreshing = false, viewAction = UpdateBalances(Adapter.Data(null, balances))) }
+                val showBanner = settingsHandler.showOwnerBanner && !ownerCredentialsRepository.hasCredentials()
```

Comment:
I think we need something separate to detect whether the user never imported (and possibly deleted) an owner. Since the button after adding a safe also depends on it, we could put it in that future PR.

Reply:
When using the SettingsHandler to keep track if an owner has been added in the past, the we don't need the ownerCredentialsRepository here.

Reply:
after updating installed app with already imported owner key you won't be able to rely just on SettingsHandler and prefs. ownerCredentialsRepository has to be checked explicitly